tket2-py/src/circuit.rs

```diff
@@ -47,20 +48,26 @@ pub fn module(py: Python) -> PyResult<&PyModule> {
 
 /// Run the validation checks on a circuit.
 #[pyfunction]
-pub fn validate_hugr(c: Py<PyAny>) -> PyResult<()> {
-    try_with_hugr(c, |hugr| hugr.validate(&REGISTRY))
+pub fn validate_hugr(c: &PyAny) -> PyResult<()> {
+    try_with_hugr(c, |hugr, _| hugr.validate(&REGISTRY))
 }
 
 /// Return a Graphviz DOT string representation of the circuit.
 #[pyfunction]
-pub fn to_hugr_dot(c: Py<PyAny>) -> PyResult<String> {
-    with_hugr(c, |hugr| hugr.dot_string())
+pub fn to_hugr_dot(c: &PyAny) -> PyResult<String> {
+    with_hugr(c, |hugr, _| hugr.dot_string())
 }
 
-/// Downcast a python object to a [`Hugr`].
+/// Cast a python tket1 circuit to a [`T2Circuit`].
 #[pyfunction]
-pub fn to_hugr(c: Py<PyAny>) -> PyResult<T2Circuit> {
-    with_hugr(c, |hugr| hugr.into())
+pub fn tket1_to_tket2(c: &PyAny) -> PyResult<T2Circuit> {
```

Comment:
these function names are very easy to confuse, in python I think `obj.to_tket1()` and `obj.to_tket2()` might be nicer?

Reply:
We cant add a `tket1.Circuit.to_tket2`, but I'll drop these and leave
```rust
Tk2Circuit(&tket1) -> Self
Tk2Circuit.to_tket1(&self) -> tket1
```

Reply:
well...it's python so you probably can add it with some horrible hacking. But i'm happier with this.